front/app/components/FormBuilder/edit/index.tsx

```diff
@@ -147,6 +152,23 @@ const FormEdit = ({
     reset,
   } = methods;
 
+  // ToDo: Would be better to find another way to achieve this, but I wasn't able to find a solution
+  // which didn't result in the form builder UI refreshing on every save.
+  useEffect(() => {
+    // Reset the form when the persisted default values change
+    // This is only run twice:
+    // 1. When the form is first opened and we need to wait for any matrix statements to be fetched
+    // and
+    // 2. When the form is changed/saved and we need to wait again for any matrix statements to be fetched
+    if (
+      defaultValues.customFields.some(
+        (field) => field.input_type === 'matrix_linear_scale'
+      )
+    ) {
+      reset(defaultValues);
+    }
+  }, [defaultValues, reset]);
+
```

Comment:
So the issue we discussed actually _wasn't_ solved by the BE in the end, so I revisited it again.

I've spent almost 2 full days on this now, and this is the only thing I've come up with which solves the issue. It causes a single extra re-render when the form is saved if there are matrix questions, but otherwise I don't think it causes too much of a problem. 

I'd still like to try and dig deeper into this issue to get it solved, but I'm thinking maybe for now to leave this in so I can finish up the other back office things (survey results, sense-making, etc.) and then re-visit this again next week? :thinking: